### PR TITLE
sql: add partitioning to compatibility

### PR DESCRIPTION
**Pull request by morgo**
Relates to https://github.com/pingcap/docs/issues/662

### CODE REVIEW

**Comment by lilin90** [comment]
@shenli PTAL

File: sql/mysql-compatibility.md [hunk 1]
```diff
@@ -140,6 +140,10 @@ Create Table: CREATE TABLE `t1` (
 
 Architecturally, TiDB does support a similar storage engine abstraction to MySQL, and user tables are created in the engine specified by the [`--store`](server-command-option.md#--store) option used when you start tidb-server (typically `tikv`).
 
+### Partitioning
+
+TiDB supports only [`RANGE` Partitioning](https://dev.mysql.com/doc/refman/5.7/en/partitioning-range.html). Support for `LIST`, `HASH`, `KEY` and subpartitioning is not available at this time.
```
Comment (shenli) [inline sql/mysql-compatibility.md]: We should mention that partitioning is not enabled by default. See https://github.com/pingcap/tidb/blob/master/sessionctx/variable/tidb_vars.go#L110
/cc @tiancaiamao

Comment (tiancaiamao) [reply sql/mysql-compatibility.md]: I believe we can enable range partition by default on the master branch within this month.

**Comment by morgo** [comment]
I will close this for now.  We can revisit with hash partitioning (coming soon).

**Pull request closed without merging**